opencensus/tags/scoped_tags.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef OPENCENSUS_TAGS_SCOPED_TAGS_H_
+#define OPENCENSUS_TAGS_SCOPED_TAGS_H_
+
+#include <initializer_list>
+#include <utility>
+
+#include "absl/strings/string_view.h"
+#include "opencensus/tags/tag_key.h"
+#include "opencensus/tags/tag_map.h"
+
+namespace opencensus {
+namespace tags {
+
+// ScopedTags is an RAII object, only ever stack-allocate it. While it's in
+// scope, the current Context will execute with the specified tags added. If a
+// tag key was already in use, its value will be replaced.
+class ScopedTags {
+ public:
+  explicit ScopedTags(
```

Comment:
Looking at other languages:
https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/tags/TagContextBuilder.java
https://github.com/census-instrumentation/opencensus-go/blob/master/tag/map.go

It seems to me that we need some possible mutations:
* upsert/put which adds if not exist + update if exists.
* remove which removes a tag.

Should we support these operations? Should we also offer them to modify a TagMap (not in the context)?

Reply:
ScopedTags does upsert.

We don't currently have a way to delete, and TagMap's API is pretty limited (I'm working on this)